### PR TITLE
fix: declare v0 endpoint added

### PR DESCRIPTION
**Pull request by Mohiiit**
This PR updates the bootstraper to ensure compatibility with Madara's distinct endpoint for declare v0.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ name: Workflow - Pull Request
 on:
   workflow_dispatch:
   pull_request:
-    branches: [main]
```
Comment (apoorvsadana) [inline .github/workflows/pull-request.yml:7]: should we undo this?

File: Dockerfile [hunk 2]
```diff
@@ -1,73 +1,206 @@
...
+RUN curl https://binaries.soliditylang.org/linux-amd64/solc-linux-amd64-v0.6.12+commit.27d51765 -o /usr/local/bin/solc-0.6.12
+RUN echo 'f6cb519b01dabc61cab4c184a3db11aa591d18151e362fcae850e42cffdfb09a /usr/local/bin/solc-0.6.12' | sha256sum --check
+RUN chmod +x /usr/local/bin/solc-0.6.12
+RUN npm install -g --unsafe-perm ganache-cli@6.12.2
+
+# First run setup-linux
+RUN make setup-linux
+
+# Then build legacy contracts
+RUN cd lib/starkgate-contracts-old && \
```
Comment (apoorvsadana) [inline Dockerfile:106]: can you add a comment here that you copied from Makefile because it make command was it working

File: Dockerfile [hunk 3]
```diff
@@ -1,73 +1,206 @@
...
+    ls -la /app/.cairo/cairo/bin && \
+    export PATH="/app/.cairo/cairo/bin:$PATH" && \
+    echo $PATH && \
+    which starknet-compile && \
+    make starkgate-contracts-latest && \
+    make braavos-account-cairo && \
+    make argent-contracts-starknet
+
+# Build the Rust project with specific binary name
+RUN cargo build --release --workspace --bin karnot-bridge-deploy
```
Comment (apoorvsadana) [inline Dockerfile:190]: let's change the binary name also to madara-bootstrapper, looks like we forgot to change this long before

File: Makefile [hunk 4]
```diff
@@ -75,38 +165,80 @@ starkgate-contracts-legacy:
...
-	cd ./lib/argent-contracts-starknet && \
- 	git checkout $(ARGENT_CONTRACTS_COMMIT_HASH) && \
- 	~/.asdf/installs/scarb/2.6.3/bin/scarb build
+	@if [ "$$(uname)" = "Darwin" ] && command -v brew >/dev/null 2>&1; then \
+		. "$$(brew --prefix asdf)/libexec/asdf.sh" && \
+		cd ./lib/argent-contracts-starknet && \
+		git checkout $(ARGENT_CONTRACTS_COMMIT_HASH) && \
+		asdf install scarb 2.6.3 && \
+		asdf local scarb 2.6.3 && \
+		scarb build; \
```
Comment (apoorvsadana) [inline Makefile:209]: can this part be moved outside because it's common

File: Makefile [hunk 5]
```diff
@@ -75,38 +165,80 @@ starkgate-contracts-legacy:
...
+	else \
+		. "$(HOME_DIR)/.asdf/asdf.sh" && \
+		cd ./lib/braavos-account-cairo && \
+		git checkout $(BRAAVOS_CONTRACTS_COMMIT_HASH) && \
+		. "$(HOME_DIR)/.asdf/asdf.sh" && \
+		asdf install scarb 2.8.4 && \
+		. "$(HOME_DIR)/.asdf/asdf.sh" && \
+		asdf local scarb 2.8.4 && \
+		scarb build; \
```
Comment (apoorvsadana) [inline Makefile:190]: can this part be moved outside because it's common

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (Mohiiit) [reply .github/workflows/pull-request.yml:7]: yes will do, was here to test out that the things are working fine

File: Makefile (hunk 5 above)
Comment (Mohiiit) [reply Makefile]: done

File: Makefile (hunk 4 above)
Comment (Mohiiit) [reply Makefile]: done

File: Dockerfile (hunk 3 above)
Comment (Mohiiit) [reply Dockerfile]: done

File: Dockerfile (hunk 2 above)
Comment (Mohiiit) [reply Dockerfile:111]: added

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (apoorvsadana) [reply .github/workflows/pull-request.yml:7]: pending

Comment (Mohiiit) [reply .github/workflows/pull-request.yml]: done

**Review by apoorvsadana — APPROVED** [review]

**Review by ocdbytes — APPROVED** [review]

**Pull request merged**